doc/developers/maintainer.rst

```diff
@@ -1,14 +1,31 @@
 Maintainer / core-developer information
 ========================================
 
+A major release, incrementing the minor version, or a big fix release
+incrementing the patch version. Our convention is that we release one or more
+release candidates (0.XXrcN) before releasing the final distributions. Creating
+the first RC involves creating the release branch (0.99.X for instance)
+directly on the main repo. The rest of the changes should be done as a PR to
```

Comment:
Could we call this stage "Freeze". Supposidely, we should not add API change, new feature after this stage.